include/manifold/common.h

```diff
@@ -13,59 +13,66 @@
 // limitations under the License.
 
 #pragma once
-#define GLM_ENABLE_EXPERIMENTAL  // needed for glm/gtx/compatibility.hpp
-#define GLM_FORCE_EXPLICIT_CTOR
-#include <glm/ext/matrix_transform.hpp>
-#include <glm/glm.hpp>
-#include <glm/gtc/constants.hpp>
-#include <glm/gtx/compatibility.hpp>
-#include <glm/gtx/rotate_vector.hpp>
 #include <limits>
 #include <stdexcept>
 #include <vector>
 
+#include "linalg.h"
+
 namespace manifold {
 /** @defgroup Math data structure definitions
- *  @brief Abstract away from glm.
+ *  @brief Abstract away from linalg.
  *  In the future the underlying data type can change.
  *  @{
  */
-using vec2 = glm::dvec2;
-using vec3 = glm::dvec3;
-using vec4 = glm::dvec4;
-using mat2 = glm::dmat2;
-using mat2x3 = glm::dmat2x3;
-using mat2x4 = glm::dmat2x4;
-using mat3x2 = glm::dmat3x2;
-using mat3 = glm::dmat3;
-using mat3x4 = glm::dmat3x4;
-using mat4x3 = glm::dmat4x3;
-using mat4 = glm::dmat4;
-using ivec2 = glm::vec<2, int>;
-using ivec3 = glm::vec<3, int>;
-using ivec4 = glm::vec<4, int>;
-using quat = glm::dquat;
+namespace la = linalg;
+using vec2 = la::vec<double, 2>;
+using vec3 = la::vec<double, 3>;
+using vec4 = la::vec<double, 4>;
+using bvec4 = la::vec<bool, 4>;
+using mat2 = la::mat<double, 2, 2>;
+using mat3x2 = la::mat<double, 3, 2>;
+using mat4x2 = la::mat<double, 4, 2>;
+using mat2x3 = la::mat<double, 2, 3>;
+using mat3 = la::mat<double, 3, 3>;
+using mat4x3 = la::mat<double, 4, 3>;
+using mat3x4 = la::mat<double, 3, 4>;
+using mat4 = la::mat<double, 4, 4>;
+using ivec2 = la::vec<int, 2>;
+using ivec3 = la::vec<int, 3>;
+using ivec4 = la::vec<int, 4>;
+using quat = la::vec<double, 4>;
 ///@}
 
+constexpr double kPi = 3.14159265358979323846264338327950288;
+constexpr double kTwoPi = 6.28318530717958647692528676655900576;
+constexpr double kHalfPi = 1.57079632679489661923132169163975144;
```

Comment:
does this seem like a reasonable place to define these?

Reply:
I think `cmath` should have `M_PI` and `M_PI_2` defined, but anyway this should be fine.

Reply:
Oof, well an hour later I know understand painfully well to never use `M_PI`.